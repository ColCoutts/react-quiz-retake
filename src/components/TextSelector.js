import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text, bottomText, image, handleChange }) {
  const imageStyle = {
    display: 'block',
    height: '250px',
    width: '250px'
  };
  
  return (
    <>
      <input name="text" value={text} onChange={handleChange} />
      <input style={imageStyle} name="image" type="file" value={image} onChange={handleChange} />
      <input name="bottomText" value={bottomText} onChange={handleChange} />
    </>
  );
}

TextSelector.propTypes = {
  image: PropTypes.string.isRequired,
  bottomText: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
}

export default TextSelector;
