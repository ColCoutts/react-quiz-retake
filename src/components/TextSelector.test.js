import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('text selector snapshot', () => {
  it('renders textselector', () => {
    const text = 'hello';
    const bottomText = 'there';
    const image = '/image';
    const wrapper = shallow(<TextSelector text={text} bottomText={bottomText} image={image} handleChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
