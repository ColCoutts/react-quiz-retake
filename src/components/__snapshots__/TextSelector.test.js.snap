// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text selector snapshot renders textselector 1`] = `
<Fragment>
  <input
    name="text"
    onChange={[Function]}
    value="hello"
  />
  <input
    name="image"
    onChange={[Function]}
    style={
      Object {
        "height": "250px",
        "width": "250px",
      }
    }
    type="file"
    value="/image"
  />
  <input
    name="bottomText"
    onChange={[Function]}
    value="there"
  />
</Fragment>
`;
